Sources/Hummingbird/Server/Request.swift

```diff
@@ -15,6 +15,20 @@
 import HummingbirdCore
 
 extension HBRequest {
+    /// Collapse body into one ByteBuffer.
+    ///
+    /// This will store the collated ByteBuffer back into the request so is a mutating method. If
+    /// you don't need to store the collated ByteBuffer on the request then use
+    /// `request.body.collate(maxSize:)`.
+    ///
+    /// - Parameter context: request context
+    /// - Returns: Collated body
+    public mutating func collateBody(context: some HBBaseRequestContext) async throws -> ByteBuffer {
+        let byteBuffer = try await self.body.collate(maxSize: context.maxUploadSize)
```

Comment:
One thing I've started wondering; in most cases we'll know the body's size through content-length. Should we leverage that information in any way?

Reply:
When you are collating the request body you could use it to compare the actual body size with the content-length and return a badRequest if they differ?